### PR TITLE
EOS-20902: SimpleProcess().run() produces incorrect output/ does not split the cmd string correctly

### PR DESCRIPTION
**Pull request by vastradparayya**
Signed-off-by: Parayya Vastrad <parayya.vastrad@seagate.com>

# Problem Statement
- SimpleProcess().run() produces incorrect output/ does not split the cmd string correctly
- Previous output split was -
cmd = '/usr/bin/ipmisimtool sdr type "Fan"'
['/usr/bin/ipmisimtool', 'sdr', 'type', '"Fan"']
- Present split output -
cmd = '/usr/bin/ipmisimtool sdr type "Fan"'
['/usr/bin/ipmisimtool', 'sdr', 'type', 'Fan']

# Design
-  https://jts.seagate.com/browse/EOS-20902

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [x] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [ ] Jira is updated
- [ ] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/utils/process.py [hunk 1]
```diff
@@ -45,8 +45,9 @@ def run(self, **args):
         for key, value in args.items():
             setattr(self, key, value)
 
+        import shlex
         try:
-            cmd = self._cmd.split() if type(self._cmd) is str else self._cmd
```
Comment (vastradparayya) [inline py-utils/src/utils/process.py:49]: Previous output was - 
cmd = '/usr/bin/ipmisimtool sdr type "Fan"'
['/usr/bin/ipmisimtool', 'sdr', 'type', '"Fan"']

File: py-utils/src/utils/process.py [hunk 2]
```diff
@@ -45,8 +45,9 @@ def run(self, **args):
...
+            cmd = shlex.split(self._cmd) if isinstance(self._cmd, str) else self._cmd
```
Comment (vastradparayya) [inline py-utils/src/utils/process.py:50]: Present output - 
cmd = '/usr/bin/ipmisimtool sdr type "Fan"'
['/usr/bin/ipmisimtool', 'sdr', 'type', 'Fan']

File: py-utils/src/utils/process.py [hunk 3]
```diff
@@ -45,8 +45,9 @@ def run(self, **args):
         for key, value in args.items():
             setattr(self, key, value)
 
+        import shlex
```
Comment (ujjwalpl) [inline py-utils/src/utils/process.py:48]: Is shlex a new dependency ? If so, we need to avoid adding new dependency.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/process.py (hunk 3 above)
Comment (vastradparayya) [reply py-utils/src/utils/process.py:48]: No @ujjwalpl . It is inbuilt library

**Review by rahul-tripathi-git — CHANGES_REQUESTED** [review]
Kindly cater PipedProcess class as well with the similar (shlex) splitting.

**Comment by vastradparayya** [comment]
> Kindly cater PipedProcess class as well with the similar (shlex) splitting.

@rahul-tripathi-git Done.

**Review by rahul-tripathi-git — APPROVED** [review]
Looks good to me.

File: py-utils/src/utils/process.py [hunk 4]
```diff
@@ -16,9 +16,11 @@
 # please email opensource@seagate.com or cortx-questions@seagate.com.
 
 import errno
+import shlex
```
Comment (sachinpunadikar) [inline py-utils/src/utils/process.py:19]: Is this a new package ?

Comment (vastradparayya) [reply py-utils/src/utils/process.py:19]: No @sachinpunadikar , It is in built package in python3.

**Review by sachinpunadikar — APPROVED** [review]
+1

**Review by ujjwalpl — APPROVED** [review]

**Pull request merged**